internal/netif/netif.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netif
+
+import (
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"golang.org/x/xerrors"
+	"k8s.io/klog"
+)
+
+type Handle interface {
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	AddrDel(link netlink.Link, addr *netlink.Addr) error
+	AddrList(link netlink.Link, family int) ([]netlink.Addr, error)
+	LinkAdd(link netlink.Link) error
+	LinkByName(name string) (netlink.Link, error)
+	LinkDel(link netlink.Link) error
+}
+
+// Manager ensures that the dummy device is created or removed.
+type Manager interface {
+	EnsureDummyDevice() error
+	RemoveDummyDevice() error
+}
+
+// netifManagerDefault is the default implementation handling creating
+// and removing of the dummy interface.
+type netifManagerDefault struct {
+	Handle
+	addr    *netlink.Addr
+	devName string
+}
+
+// NewNetifManager returns a new instance of NetifManager with the ip address set to the provided values
+// These ip addresses will be bound to any devices created by this instance.
+func NewNetifManager(ip net.IP, devName string) Manager {
+	return &netifManagerDefault{
+		&netlink.Handle{},
+		&netlink.Addr{
+			IPNet: netlink.NewIPNet(ip),
+			// We only want the scope to be host
+			Scope: 0xfe,
```

Comment:
can you use `unix. RT_SCOPE_HOST` instead?

Reply:
The constant applies only for linux. There is no such thing on `darwin` or `windows`. And it makes the compilation impossible.